app/jobs/reports/base_report.rb

```diff
@@ -62,10 +62,10 @@ def upload_file_to_s3_timestamped_and_latest(report_name, body, extension)
       url
     end
 
-    def generate_s3_paths(name, extension, now: Time.zone.now)
+    def generate_s3_paths(name, extension, subname: nil, now: Time.zone.now)
       host_data_env = Identity::Hostdata.env
-      latest = "#{host_data_env}/#{name}/latest.#{name}.#{extension}"
-      [latest, "#{host_data_env}/#{name}/#{now.year}/#{now.strftime('%F')}.#{name}.#{extension}"]
+      latest = "#{host_data_env}/#{name}/latest.#{name}#{subname ? '/' + subname : '' }.#{extension}"
+      [latest, "#{host_data_env}/#{name}/#{now.year}/#{now.strftime('%F')}.#{name}#{subname ? '/' + subname : '' }.#{extension}"]
```

Comment:
rubocop is gonna get mad at this line length, maybe we pull the `subname` but into a `subdirectory`
```suggestion
      name_subdir_ext = "#{name}#{subdir}.#{extension}"
      latest = "#{host_data_env}/#{name}/latest.#{name_subdir_ext}"
      [latest, "#{host_data_env}/#{name}/#{now.year}/#{now.strftime('%F')}.#{name_subdir_ext}"]
```